include/alpaka/workdiv/WorkDivMembers.hpp

```diff
@@ -33,6 +35,33 @@ namespace alpaka
         {
         }
 
+        //! \brief Accepts single specific type and is called without explicit template parameters.
+        ALPAKA_NO_HOST_ACC_WARNING
+        ALPAKA_FN_HOST_ACC explicit WorkDivMembers(
+            alpaka::Vec<TDim, TIdx> const& gridBlockExtent,
+            alpaka::Vec<TDim, TIdx> const& blockThreadExtent,
+            alpaka::Vec<TDim, TIdx> const& elemExtent)
+            : m_gridBlockExtent(gridBlockExtent)
+            , m_blockThreadExtent(blockThreadExtent)
+            , m_threadElemExtent(elemExtent)
+        {
+        }
+
+        //! \brief Accepts single specific type and is called without explicit template parameters.
+        //! \tparam T is the element type of the array
+        //! \tparam N is the size of the array
+        template<class T, size_t N>
+        ALPAKA_FN_HOST_ACC explicit WorkDivMembers(
+            std::array<T, N> const gridBlockExtent,
```

Comment:
The goal should not be to add a constructor which understand `std::array`. We like to stay interpotable with any alpaka foreign type which represents extents. `std::array` is one example it could also be `std::vector` or users type like in PIConGPU `pmacc::math::Vector<>`. 
To reach this goal we have the traits these give the alpaka user the possibility to be specialize for there own types.